### PR TITLE
Support modifying RRsets

### PR DESCRIPTION
**Pull request by dasJ**

### CODE REVIEW

File: files/delete-rrsets [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# Parameters
+url="${1}/api/v1/servers/${2}"
+# $2 is the server
+apikey="${apikey:-}"
+zonename="${3}"
+# All RRsets are passed via an env variable because the quotes are mangled when
+# specifying the JSON on the command line.
+rrsets="${rrsets:-}"
+
+rrsets='[{"records": [{"content": "1.2.3.4", "disabled": false, "set-ptr": false}, {"content": "5.6.7.8"}], "type": "A", "name": "efcloud.co", "ttl": 500}, {"records": [{"content": "4.5.6.7"}], "type": "A", "name": "*.efcloud.co", "ttl": 4000}]'
```
Comment (Scriptkiddi) [inline files/delete-rrsets]: remove this

File: files/upsert-rrset [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+# Parameters
+url="${1}/api/v1/servers/${2}"
+# $2 is the server
+apikey="${apikey:-}"
+zonename="${3}"
+# The rrset is passed via an env variable because the quotes are mangled when
+# specifying the JSON on the command line.
+expected="${rrset:-}"
+
+# Ensure every record has a "disabled" field
```
Comment (Scriptkiddi) [inline files/upsert-rrset]: change this comment to something that describes this better

**Review by Scriptkiddi — DISMISSED** [review]

File: tasks/single-zone.yml [hunk 3]
```diff
@@ -32,3 +32,34 @@
   with_dict: "{{ item.metadata | default({'SOA-EDIT': ''}) }}"
   loop_control:
     loop_var: meta
+
+- name: Upsert RRsets
+  command:
+    cmd: >
+      /usr/local/bin/pdns-auth-api-upsert-rrset "{{ pdns_auth_api_connect }}" "{{ pdns_auth_api_server }}"
+      "{{ item.name }}"
```
Comment (SlothOfAnarchy) [inline tasks/single-zone.yml:40]: why don't you use `loop_control` for this element? you could then use `zone.name` instead

File: tasks/single-zone.yml [hunk 4]
```diff
@@ -32,3 +32,34 @@
...
+  with_items: "{{ item.rrsets }}"
+  loop_control:
+    loop_var: rrset
+
+- name: Remove unknown RRsets
+  command:
+    cmd: >
+      /usr/local/bin/pdns-auth-api-delete-rrsets "{{ pdns_auth_api_connect }}" "{{ pdns_auth_api_server }}"
+      "{{ item.name }}"
+
```
Comment (SlothOfAnarchy) [inline tasks/single-zone.yml]: please remove this gap

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]
see comments

File: tasks/single-zone.yml (hunk 3 above)
Comment (dasJ) [reply tasks/single-zone.yml:40]: Because it works like this and more code menas more comlexity

Comment (SlothOfAnarchy) [reply tasks/single-zone.yml:40]: but it would also make it easier to understand the code

**Comment by dasJ** [comment]
@SlothOfAnarchy Pls rereview

**Review by SlothOfAnarchy — APPROVED** [review]

**Pull request merged**